lib/cucumber/cli/rerun_file.rb

```diff
@@ -0,0 +1,29 @@
+module Cucumber
+  module Cli
+    class RerunFile
+      attr_reader :path
+
+      def self.can_read?(path)
+        path[0] == '@' && File.file?(real_path(path))
+      end
+
+      def self.real_path(path)
+        path[1..-1] # remove leading @
+      end
+
+      def initialize(path)
+        @path = self.class.real_path(path)
+      end
+
+      def features
+        lines.map { |l| l.scan(/(?:^| )(.*?\.feature(?:(?::\d+)*))/) }.flatten
```

Comment:
@pasku - agree that looks cleaner. The issue is this only flattens the arrays produced by `lines` and `scan`, but there are also captures which produce arrays and this doesn't deal with those (I think :confused:) ...can you think of anything else?